gr-digital/lib/crc32_bb_impl.cc

```diff
@@ -65,21 +65,40 @@ namespace gr {
       }
     }
 
+    unsigned int
+    crc32_bb_impl::calculate_crc32(const unsigned char* in, size_t packet_length){
+      unsigned int crc = 0;
+      d_crc_impl.reset();
+      if (!d_packed){
+        const size_t n_packed_length = 1 + ((packet_length - 1) / 8);
+        unsigned char packed_buffer[n_packed_length];
+        memset(packed_buffer, 0, n_packed_length);
+        for (size_t bit = 0; bit < packet_length; bit++){
+          packed_buffer[bit/8] |= (in[bit] << (bit % 8));
```

Comment:
We should really have a VOLK kernel for this. **should.** Feels like I thought we had a "packer" kernel; @n-west ?

Nothing wrong with this code, though! 😄 

Thinking about that, we should probably just use a different CRC polynomial (this polynomial g(x) "interpolated" by doing g<sup>8</sup>(x), maybe?) and not pack before CRC. Anyway, that's math for another day. (Which doesn't mean I wouldn't encourage you to do that math or just try. Just saying my own brain's too mushy right now.)

Reply:
Ah jetzt ja. (eine Insel.)

So, indeed, 

```
    boost::crc_optimal<32,
                       boost::crc_32_type::truncated_polynominal * boost::crc_32_type::truncated_polynominal *
                           boost::crc_32_type::truncated_polynominal * boost::crc_32_type::truncated_polynominal *
                           boost::crc_32_type::truncated_polynominal * boost::crc_32_type::truncated_polynominal *
                           boost::crc_32_type::truncated_polynominal * boost::crc_32_type::truncated_polynominal,
                       0xFFFFFFFF, 0xFFFFFFFF, true, true>
        padded_crc;
```

at least in my stupid test case works directly on unpacked data and delivers the same output as `crc_32_type` (which is what the crc32_impl uses) on packed data.

Why the ugly unrolled potentiation? The polynomial is a template argument, i.e. needs to be a compile-time constant. Prior to C++11, there's not `constexpr` to let me define a compile-time evaluated function, and prior to C++14, you couldn't use a `for` loop in that function.